### ISSUE TITLE
codebuild project override artifact name in spec file

### ISSUE THREAD

**Issue by ranga543** [terraform-providers/terraform-provider-aws#5908]
### Description

Allow codebuild project resource to specify `override_artifact_name` option in `artifacts` section.

### New or Affected Resource(s)

* aws_codebuild_project

### Potential Terraform Configuration

```hcl
"override_artifact_name": {
	Type:     schema.TypeBool,
	Optional: true,
	Default:  false,
},
```

### References

* [docs](https://github.com/aws/aws-sdk-go/blob/master/service/codebuild/api.go#L3640)
* add new property after [this line](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_codebuild_project.go#L37) called `override_artifact_name`

**Comment by jckuester** [comment]
PR to add this feature: https://github.com/terraform-providers/terraform-provider-aws/pull/7824

**Comment by stuartpurgavie** [comment]
Is there anything we can do to move this up the review & merge priority list?

**Comment by bflad** [comment]
@stuartpurgavie it looks like this issue was created before we had a common heading for feature requests:

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

---

Related to the first item, the maintainers can use 👍 reactions on issues and pull requests as a priority for handling them. e.g. https://github.com/terraform-providers/terraform-provider-aws/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by bflad** [comment]
Support for a new `override_artifact_name` argument under the `artifacts` and `secondary_artifacts` configuration blocks has been merged and will release with version 2.22.0 of the Terraform AWS Provider, later this week. Thanks to @jckuester for the implementation. 👍
